### PR TITLE
Add types for new appdynamics package

### PR DESCRIPTION
**Pull request by justinvos**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
what's about this project address and the way it's registered on NPM?
what's the use of single method? (why process is the most important in your opinion?). This package installs a lot of stuff and uses post install scripts

**Comment by justinvos** [comment]
> what's about this project address and the way it's registered on NPM?
AppDynamics is a proprietary product from Cisco so I assume that's why they don't link to their repository on their NPM package info.

> what's the use of single method? (why process is the most important in your opinion?). This package installs a lot of stuff and uses post install scripts
`profile` is the main function to begin the monitoring and logging to the AppDynamics back-end systems. I have seen there is more functions available but 99% of the use-cases of this package is just to run this profile function.

**Comment by justinvos** [comment]
> what's about this project address and the way it's registered on NPM?

Is there a better way to specify that there's no public link to the repository?

**Comment by peterblazejewicz** [comment]
This should point to project/service page, e.g., to avoid giving impression of kind of scam, sorry:
```ts
// Type definitions for non-npm package Microsoft Access 14.0 Object Library - Access 14.0
// Project: https://msdn.microsoft.com/en-us/library/dn142571.aspx
```
```ts
// Type definitions for non-npm package clearbladejs-node 1.0
// Project: https://github.com/ClearBlade/Node-SDK
```

File: types/appdynamics/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Type definitions for appdynamics 21.8
+// Project: https://github.com/npm/npm
```
Comment (weswigham) [inline types/appdynamics/index.d.ts:2]: Agree, this should be changed to some kind of project link, even if it's not the project source code.

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/appdynamics/index.d.ts (hunk 1 above)
Comment (justinvos) [reply types/appdynamics/index.d.ts]: No worries at all - I've updated it to point to https://www.appdynamics.com/supported-technologies/nodejs

Is this the sort of page you're thinking?

The other page that could work in my mind is:
https://docs.appdynamics.com/21.6/en/application-monitoring/install-app-server-agents/node-js-agent

Comment (justinvos) [reply types/appdynamics/index.d.ts]: Hi @weswigham! Are you able to re-review this PR? It think it's ready now

**Review by weswigham — APPROVED** [review]

**Comment by justinvos** [comment]
Ready to merge

File: types/appdynamics/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * appdynamics monitors your Node.js applications in production.
+ */
+export as namespace appdynamics;
+
+export interface ProfileParameter {
+  controllerHostName?: string;
+  controllerPort?: number;
+  controllerSslEnabled?: boolean;
+  tierName?: string;
+  nodeName?: string;
```
Comment (shrinivas91) [inline types/appdynamics/index.d.ts]: Can we add more fields in here? like accountAccessKey, applicationName, reuseNode, reuseNodePrefix and logging?

**Pull request merged**